Scripts/Source/minai_Arousal.psc

```diff
@@ -44,43 +59,70 @@ function Maintenance(minai_MainQuestController _main)
+    SLA_KillerHeels = Game.GetFormFromFile(0x08F326, "SexlabAroused.esm") as Keyword
+    SLA_PantsNormal = Game.GetFormFromFile(0x08F3F3, "SexlabAroused.esm") as Keyword
+    SLA_MicroHotPants = Game.GetFormFromFile(0x08F3F4, "SexlabAroused.esm") as Keyword
+    SLA_ThongGstring = Game.GetFormFromFile(0x08F3F5, "SexlabAroused.esm") as Keyword
+    SLA_ArmorHarness = Game.GetFormFromFile(0x08F3FC, "SexlabAroused.esm") as Keyword
+    SLA_ArmorTransparent = Game.GetFormFromFile(0x08F3FD, "SexlabAroused.esm") as Keyword
+    SLA_ArmorLewdLeotard = Game.GetFormFromFile(0x08F401, "SexlabAroused.esm") as Keyword
+    SLA_PelvicCurtain = Game.GetFormFromFile(0x08F402, "SexlabAroused.esm") as Keyword
+    SLA_FullSkirt = Game.GetFormFromFile(0x08F40D, "SexlabAroused.esm") as Keyword
+    SLA_MiniSkirt = Game.GetFormFromFile(0x08F40E, "SexlabAroused.esm") as Keyword
+    SLA_MicroSkirt = Game.GetFormFromFile(0x08F40F, "SexlabAroused.esm") as Keyword
+    SLA_BootsHeels = Game.GetFormFromFile(0x08F410, "SexlabAroused.esm") as Keyword
+    SLA_HasLeggings = Game.GetFormFromFile(0x08FE9F, "SexlabAroused.esm") as Keyword
+    SLA_ArmorRubber = Game.GetFormFromFile(0x08FEA4, "SexlabAroused.esm") as Keyword
+    EroticArmor = Game.GetFormFromFile(0x08C7F6, "SexlabAroused.esm") as Keyword
+    SLA_PiercingVulva = Game.GetFormFromFile(0x08F3F6, "SexlabAroused.esm") as Keyword
+    SLA_PiercingBelly = Game.GetFormFromFile(0x088F3F7, "SexlabAroused.esm") as Keyword
+    SLA_PiercingNipple = Game.GetFormFromFile(0x08F3F8, "SexlabAroused.esm") as Keyword
+    SLA_PiercingClit = Game.GetFormFromFile(0x08F40B, "SexlabAroused.esm") as Keyword
+    
     ; Check a couple keywords to see if it's a stripped down SexlabAroused
     if SLA_HalfNakedBikini && SLA_ArmorHalfNaked
       bHasArousedKeywords = True
     EndIf
     Main.Info("Sexlab Aroused Keywords=" + bHasArousedKeywords)
   EndIf
-  if Game.GetModByName("OSLAroused.esp") != 255
-    Main.Info("Found OSL Aroused")
-    bHasOSL = True
+
+  if Game.GetModByName("TheNewGentleman.esp") != 255
+    bHasTNG = True
+	Main.Info("Found TNG")
+	TNG_XS = Game.GetFormFromFile(0x03BFE1, "TheNewGentleman.esp") as Keyword
+	TNG_S = Game.GetFormFromFile(0x03BFE2, "TheNewGentleman.esp") as Keyword
+	TNG_M = Game.GetFormFromFile(0x03BFE3, "TheNewGentleman.esp") as Keyword
+	TNG_L = Game.GetFormFromFile(0x03BFE4, "TheNewGentleman.esp") as Keyword
+	TNG_XL = Game.GetFormFromFile(0x03BFE5, "TheNewGentleman.esp") as Keyword
+	TNG_DefaultSize = Game.GetFormFromFile(0x03BFE0, "TheNewGentleman.esp") as Keyword
+	TNG_Gentlewoman = Game.GetFormFromFile(0x03BFF8, "TheNewGentleman.esp") as Keyword
+	TNG_Revealing = Game.GetFormFromFile(0x03BFFF, "TheNewGentleman.esp") as Keyword
+	if TNG_XS != None && TNG_S != None && TNG_M != None && TNG_L != None && TNG_XL != None && TNG_DefaultSize != None
+		Debug.Trace("[minai] TNG size keywords retrieved successfully.")
```

Comment:
I added new functions for logging in a recent update: 
Info(), Warn(), Error(), and Debug(). Call Main.Info("TNG size keywords....") instead here. Do this everywhere you have Debug.Trace right now.

By doing this, we can adjust the logging level on the fly to be a better citizen to our users. :)